app/models/piece.rb

```diff
@@ -100,4 +100,22 @@ def is_obstructed_diagonally(piece_destination)
       end 
     end 
   end 
+
+  def move_to!(x_new, y_new)
+    piece.position_x = x_current
+    piece.position_y = y_current
+    position_x = x_destination
+    position_y = y_destination
+
+    #moving to an empty space, move is valid
+    if is_occupied?(x_destination, y_destination) == false && valid_move?
```

Comment:
A few comments:
1. `if condition == false` can be refactored to `if !condition`.. `!` negates the result of a condition..
2. you'll need to pass x_new and y_new to `valid_move?`

Reply:
Made the changes!